carbonmark-api/src/utils/helpers/fetchProjectPoolInfo.ts

```diff
@@ -23,72 +24,144 @@ type PoolInfoMap = {
-  });
+  const data =
+    await gqlSdk.digital_carbon.getCarbonProjectsByProjectIDAndVintage({
+      projectID: params.projectID,
+      vintage: Number(params.vintage),
+    });
 
   /** @type {QueryResponse[]} */
-  const tokens = data?.carbonOffsets?.length ? data.carbonOffsets : [];
 
-  const stats: Stats = tokens.reduce(
+  const tokens: CarbonCredits = data?.carbonProjects?.[0]?.carbonCredits || [];
+
+  // Graph data is in 18 decimals. All operations are performed in BigNumber before converting to Number at the end
+
+  const bigNumberStats: BigNumberStats = tokens.reduce(
     (stat, token) => ({
-      totalBridged: stat.totalBridged + Number(token.totalBridged),
-      totalRetired: stat.totalRetired + Number(token.totalRetired),
-      totalSupply: stat.totalSupply + Number(token.currentSupply),
+      bridged: stat.bridged.add(ethers.BigNumber.from(token.bridged)),
+      retired: stat.retired.add(ethers.BigNumber.from(token.retired)),
+      totalSupply: stat.totalSupply.add(
+        ethers.BigNumber.from(token.currentSupply)
+      ),
     }),
     initialStats
   );
 
+  // project bigNumber stats
+  const stats: Stats = {
+    totalBridged: parseFloat(
+      ethers.utils.formatUnits(bigNumberStats.bridged, 18)
+    ),
+    totalRetired: parseFloat(
+      ethers.utils.formatUnits(bigNumberStats.retired, 18)
+    ),
+    totalSupply: parseFloat(
+      ethers.utils.formatUnits(bigNumberStats.totalSupply, 18)
+    ),
+  };
   const poolInfoMap = Object.keys(POOL_INFO).reduce<Partial<PoolInfoMap>>(
     (prevMap, poolName) => {
-      const balanceKey = `balance${poolName.toUpperCase()}`; // e.g. "balanceBCT"
-      // see if any of the entries from the subgraph have a balance > 1, otherwise return "0"
-      const matchingTokenInfo = tokens.find(
-        (t) => Number(get(t, balanceKey)) > 1
-      );
+      const poolAddress = POOL_INFO[poolName].poolAddress;
+      let totalSupply = 0;
+      let matchingTokenInfo: PoolBalance | undefined;
+
+      for (const token of tokens) {
+        const potentialMatch = token.poolBalances.find(
+          (t) => t.pool.id.toLowerCase() === poolAddress.toLowerCase()
+        );
+
+        if (potentialMatch) {
+          matchingTokenInfo = potentialMatch;
+          const decimals = potentialMatch.pool.decimals;
+          const numberBalance = parseFloat(
+            ethers.utils.formatUnits(potentialMatch.balance, decimals)
+          );
+          totalSupply += numberBalance;
+        }
+      }
+
+      if (!matchingTokenInfo) {
+        return prevMap;
+      }
 
       return {
         ...prevMap,
         [poolName]: {
           poolName,
-          supply: get(matchingTokenInfo, balanceKey, "0"),
-          poolAddress: POOL_INFO[poolName].poolAddress,
+          supply: totalSupply.toString(),
+          poolAddress,
           isPoolDefault:
-            matchingTokenInfo?.tokenAddress.toLowerCase() ===
+            matchingTokenInfo.pool.id.toLowerCase() ===
```

Comment:
Here you are comparing the pool address (e.g. BCT address) when you should be comparing `matchingTokenInfo.id` which is the address of the TCO2 or C3T

Reminder that a token can have multiple addresses, and be in multiple pools. E.g. VCS-981-2014 which has 2 assets in 3 pools.